src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -343,12 +346,18 @@ public String syncWorkspacePolicy(
    * @return
    */
   @Traced
-  public String syncResourcePolicy(
+  public String syncPrivateResourcePolicy(
       ControlledResource resource,
       ControlledResourceIamRole role,
       AuthenticatedUserRequest userReq) {
+    // TODO: in the future, this function will also be called for application managed resources,
+    //  including app-shared. This check should be modified appropriately.
+    if (resource.getAccessScope() != AccessScopeType.ACCESS_SCOPE_PRIVATE) {
```

Comment:
Right now we just throw at creation time if someone creates an application-managed resource, so it wouldn't execute either way. I'm fairly sure we could make this check "private or application-managed", but I've been wrong on applications before. I'm inclined to leave it out until we actually support application-managed resources

Reply:
I think it still would be called for application shared, because we need to grant one app SA extra (i.e. resource editor) permission on the resource. Either way, I'd rather not implement this until we have apps so we can say for sure.